hangman.rb

```diff
@@ -0,0 +1,68 @@
+
+$random_words = ['ruby', 'hangman', 'flutter', 'developer', 'omkar', 'github', 'josh', 'assignment']
+
+def choose_word
+  $random_words.sample
+end
+
+def display_word(word,correct_guesses)
+  word.chars.map { |char| correct_guesses.include?(char) ? char : '_' }.join(' ')
+end
+ 
+
+def check_game_result(word,remaining_count)
+  if remaining_count == 0
+    puts "You guessed the word '#{word}' correctly"
+    return
+  end
+  
+  puts "Better luck next time, the word was '#{word}'"
+end
+
+def update_guesses(word, guess_letter, correct_guesses, incorrect_guesses, remaining_count)
+  if word.include?(guess_letter)
+    correct_guesses.push(guess_letter)
+    remaining_count -= 1
+  else
+    incorrect_guesses.push(guess_letter)
+  end
+  remaining_count
+end
+
+def process_guess(word, guess_letter, correct_guesses, incorrect_guesses, remaining_count)
+  unless guess_letter.length == 1 && ('a'..'z').include?(guess_letter)
```

Comment:
move the conditional statements to separate method
eg:
```
def invalid_guess?(guess)
  guess.length != 1 || !('a'..'z').cover?(guess)
end

def process_guess(word, guess_letter, correct_guesses, incorrect_guesses, remaining_count)
  if valid_guess?(guess_letter)
```
1. Better readability
2. Prefer `if` over `unless`(keeps the code less confusing).
3. Use `cover?` to check inclusion in range